src/kernel/src/hv/linux/mod.rs

```diff
@@ -139,33 +139,32 @@ impl Vm {
         Ok(VCpu {
             fd,
             kvm_run: NonNull::new(kvm_run.cast()).unwrap(),
-            mmap_size,
         })
     }
 }
 
 #[derive(Debug)]
-pub struct VCpus([VCpu; 8]);
-
-#[derive(Debug)]
-struct VCpu {
-    fd: OwnedFd,
-    kvm_run: NonNull<KvmRun>,
+pub struct VCpus {
+    vcpus: [VCpu; 8],
     mmap_size: usize,
 }
 
-impl Drop for VCpu {
+impl Drop for VCpus {
     fn drop(&mut self) {
-        use libc::munmap;
-
-        unsafe {
-            if munmap(self.kvm_run.as_ptr().cast(), self.mmap_size) < 0 {
-                panic!("failed to munmap KVM_RUN: {}", Error::last_os_error());
-            };
+        for vcpu in &self.vcpus {
+            unsafe {
+                libc::munmap(vcpu.kvm_run.as_ptr().cast(), self.mmap_size);
+            }
```

Comment:
I don't think this is a good idea. Each memory map belongs to each vCPU. We will have a bug if someone move out a vCPU from this struct and doesn't realize they also need to fix this.

Reply:
Maybe you're right. Hmmm